pkg/cache/memory/kvcache.go

```diff
@@ -43,102 +40,92 @@ func NewKeyValueCache[T any](opts ...KeyValueCacheOptionFn) (*keyValueCache[T],
 	}
 
 	return &keyValueCache[T]{
-		values: make(map[string]cacheValue[T]),
+		values: xsync.NewMap[string, cacheValue[T]](),
 		config: config,
 	}, nil
 }
 
 // Get retrieves the value from the cache with the given key.
 func (c *keyValueCache[T]) Get(key string) (T, bool) {
 	var zero T
-	c.valuesMu.RLock()
-	defer c.valuesMu.RUnlock()
-
-	cachedValue, exists := c.values[key]
-	if !exists {
+	v, ok := c.values.Load(key)
+	if !ok {
 		return zero, false
 	}
-
-	isCacheValueExpired := time.Since(cachedValue.cachedAt) > c.config.ttl
```

Comment:
Any reason we're removing intermediate variables? I believe it's a net positive for readability